easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -466,34 +467,27 @@ def toolchain(self):
             self.log.debug("Initialized toolchain: %s (opts: %s)" % (tc_dict, self['toolchainopts']))
         return self._toolchain
 
+
     def dump(self, fp):
         """
         Dump this easyconfig to file, with the given filename.
         """
         eb_file = file(fp, "w")
 
-        def to_str(x):
-            """Return quoted version of x"""
-            if isinstance(x, basestring):
-                if '\n' in x or ('"' in x and "'" in x):
-                    return '"""%s"""' % x
-                elif "'" in x:
-                    return '"%s"' % x
-                else:
-                    return "'%s'" % x
-            else:
-                return "%s" % x
-
         # ordered groups of keys to obtain a nice looking easyconfig file
         grouped_keys = [
+            ['easyblock'],
             ['name', 'version', 'versionprefix', 'versionsuffix'],
             ['homepage', 'description'],
             ['toolchain', 'toolchainopts'],
-            ['source_urls', 'sources'],
+            ['sources', 'source_urls'],
             ['patches'],
             ['builddependencies', 'dependencies', 'hiddendependencies'],
             ['parallel', 'maxparallel'],
-            ['osdependencies']
+            ['osdependencies'],
+            ['prebuildopts', 'preinstallopts', 'preconfigopts'],
+            ['sanity_check_paths'],
+            ['moduleclass']
```

Comment:
don't include these two here, they should always be at the end, so write these out (with a newline in between) after the for loop that tackles the 'other' parameters